fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryLockTest.java

```diff
@@ -87,4 +89,30 @@ void testFileLock(boolean useAgile) throws IOException {
             lock2.close();
         }
     }
+
+    @ParameterizedTest
+    @CsvSource({"true,true", "true,false", "false,true", "false,false"})
+    void testFileLockClose(boolean useAgile, boolean abortAgilityContext) throws IOException {
+        try (FDBRecordContext context = fdb.openContext()) {
+            AgilityContext agilityContext =
+                    useAgile ?
+                    AgilityContext.agile(context, 1000, 100_0000) :
+                    AgilityContext.nonAgile(context);
+
+            FDBDirectory directory = new FDBDirectory(subspace, null, null, null, true, agilityContext);
+            String lockName = "file.lock";
+            final Lock lock1 = directory.obtainLock(lockName);
+            lock1.ensureValid();
+            if (abortAgilityContext) {
+                agilityContext.abortAndClose();
+                assertTrue(agilityContext.isClosed());
+            } else {
+                assertFalse(agilityContext.isClosed());
+            }
+            lock1.close();
+            directory.close();
+            agilityContext.flushAndClose();
+            context.commit();
```

Comment:
It would probably be good to validate that you can obtain a new lock, but I don't think this PR changes the risk here; this test covers that, as-is.